### PR TITLE
Fix NuRaft launcher race condition when creating raft server

### PR DESCRIPTION
**Pull request by as51340**
### Description

Fixes bug in NuRaft by removing raft_launcher and instead manually setting up raft_server, rpc_listener...
We consider raft server successfully created once it is behaving as a leader.

[master < Task] PR
- [x] Provide the full content or a guide for the final git message
    - **Fix NuRaft launcher race condition when creating raft server**

### CI Testing Labels
Please select the appropriate CI test labels _(CI -build=**build-name** -test=**test-suite**)_

### Documentation checklist
- [x] Add the documentation label tag
- [x] Add the bug / feature label tag
- [x] Add the milestone for which this feature is intended
    - If not known, set for a later milestone
- [x] Write a release note, including added/changed clauses
    - **Fix NuRaft launcher race condition when creating raft server**
- [ ] Link the documentation PR here
    - **[Documentation PR link]**
- [x] Tag someone from docs team in the comments

### CODE REVIEW

**Comment by as51340** [comment]
@kgolubic docs not needed

File: tests/unit/coordinator_instance.cpp [hunk 1]
```diff
@@ -73,6 +73,8 @@ TEST_F(CoordinatorInstanceTest, ConnectCoordinators) {
                                                           main_data_directory / "high_availability1" / "coordinator"};
 
   auto instance1 = CoordinatorInstance{init_config1};
+  while (!instance1.IsLeader()) {
+  }
```
Comment (antoniofilipovic) [inline tests/unit/coordinator_instance.cpp:77]: Do we need this still?

**Review by antoniofilipovic — CHANGES_REQUESTED** [review]
All looks good, I just have one comment

File: tests/unit/coordinator_instance.cpp (hunk 1 above)
Comment (as51340) [reply tests/unit/coordinator_instance.cpp:77]: no

**Review by antoniofilipovic — APPROVED** [review]

**Pull request merged**